In [1]:
import numpy as np
import time
import matplotlib.pyplot as plt

sizes = np.asarray([10, 100, 1e3, 1e4, 1e5, 1e6, 1e7], dtype=int)
ts = [np.linspace(0, 10, s) for s in sizes]



In [2]:
def func(t):
    return np.exp(-t)


In [3]:
for t in ts:
    %timeit func(t)


1.23 µs ± 48.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
2.08 µs ± 318 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
13.8 µs ± 2.61 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
78.2 µs ± 15.5 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
1.23 ms ± 26.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
12.8 ms ± 125 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
136 ms ± 919 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


8700 6core 12 threads

819 ns ± 1.68 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

1.29 µs ± 1.99 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

6.53 µs ± 78 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

53.6 µs ± 540 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

526 µs ± 1.27 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

7.87 ms ± 102 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

78.3 ms ± 1.26 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)


6500U 2C4T

1.23 µs ± 48.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

2.08 µs ± 318 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

13.8 µs ± 2.61 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

78.2 µs ± 15.5 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

1.23 ms ± 26.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

12.8 ms ± 125 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

136 ms ± 919 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [15]:
times = [0.8, 1.29, 6.53, 53.6, 526, 7870, 78300]

plt.plot(sizes, times)
plt.yscale('log')
plt.xscale('log')

plt.show()

AttributeError: module 'matplotlib.pyplot' has no attribute 'semilogxy'